### PR TITLE
Add AppVeyor configuration

### PR DESCRIPTION
**Pull request by Watson1978**
To support Windows environment, we can use AppVeyor CI for testing.

As the first step, It works with the following combinations.

- Ruby+Devkit 2.5.3-1 (https://rubyinstaller.org/downloads/)
- ImageMagick 6.9.10.24

Example)
https://ci.appveyor.com/project/Watson1978/rmagick/builds/21991269

@mockdeep Can you add AppVeyor integration in this repository if you interest ?

### CODE REVIEW

**Comment by mockdeep** [comment]
Hey @Watson1978, I've enabled AppVeyor. Not sure if it will run on this PR. You may need to have to force push a new commit, or if that doesn't work you may need to open a new PR.

**Comment by Watson1978** [comment]
@mockdeep Thank you so much!! :)

**Comment by mockdeep** [comment]
Hey @Watson1978 do you think you can get the build passing on Appveyor before we merge this?

**Comment by Watson1978** [comment]
@mockdeep hmm, I think we need to fix after merge. Some failures are common to other CI result, I guess...

**Comment by mockdeep** [comment]
Okay, I think it might be worth waiting for [my CircleCI branch](https://github.com/rmagick/rmagick/pull/322) to be merged once that's consistently passing, and then we can hopefully rebase this and get it passing before merging.

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+environment:
+  matrix:
+    - RUBY_VERSION: "25-x64"
```
Comment (mockdeep) [inline .appveyor.yml]: @Watson1978 do you think you can get it to pass on Ruby 2.3 and IM 6.8?

Comment (Watson1978) [reply .appveyor.yml]: I want to test that it works with https://github.com/rmagick/rmagick/pull/315 and the PR effect to Ruby 2.4+.

Ruby 2.4 and IM 6.8 is better

Comment (Watson1978) [reply .appveyor.yml]: Hmm, IM 6.8 is too old and it cause a failure to build :(
https://ci.appveyor.com/project/mockdeep/rmagick/builds/22085425

Comment (mockdeep) [reply .appveyor.yml]: Hmm, why would `GetExceptionInfo` be missing on Windows? These compile fine on Linux. Looking at the configured options, one thing that stands out to me is that it is missing `:magick_config` and `:local_libs`. Does that seem significant to you?

Appveyor:

```rb
Configured compile options: {
  :magick_config=>nil,
  :with_magick_wand=>nil,
  :pkg_config=>nil,
  :magick_version=>"6.8.9",
  :local_libs=>"",
  :cflags=>"-march=x86-64 -mtune=generic -O2 -pipe",
  :cppflags=>"-I\"C:\\Program Files\\ImageMagick-6.8.9-Q16/include\"",
  :ldflags=>"-L\"C:\\Program Files\\ImageMagick-6.8.9-Q16/lib\"",
  :defs=>[],
  :config_h=>"Makefile rmagick.h"
}
```

TravisCI

```rb
Configured compile options: {
  :magick_config=>true,
  :with_magick_wand=>false,
  :pkg_config=>false,
  :magick_version=>"6.8.9",
  :local_libs=>" -lMagickCore-6.Q16  ",
  :cflags=>" -fopenmp -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/usr/include/ImageMagick-6  ",
  :cppflags=>" -fopenmp -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/usr/include/ImageMagick-6  ",
  :ldflags=>" -lMagickCore-6.Q16  ",
  :defs=>[],
  :config_h=>"Makefile rmagick.h"
}
```

Comment (Watson1978) [reply .appveyor.yml]: By some reason, `GetExceptionInfo()` API was not contained in library of old ImageMagick.

File: spec/rmagick/image/read_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 RSpec.describe Magick::Image, '#read' do
   describe 'issue #200' do
     before do
-      pid = Process.spawn File.join(SUPPORT_DIR, 'issue_200', 'app.rb'), err: :close, out: :close
+      pid = Process.spawn "ruby #{File.join(SUPPORT_DIR, 'issue_200', 'app.rb')}", err: :close, out: :close
```
Comment (mockdeep) [inline spec/rmagick/image/read_spec.rb]: Looks like at least one failure is caused by this test. I think [we can just comment out the entire `before` block](https://github.com/rmagick/rmagick/pull/342), since both tests below are `skip`ped anyway.

File: .appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  matrix:
+    - RUBY_VERSION: "24-x64"
+
+install:
+  - appveyor DownloadFile https://ftp.icm.edu.pl/pub/graphics/ImageMagick/binaries/ImageMagick-6.8.9-10-Q16-x64-dll.exe
+  - ImageMagick-6.8.9-10-Q16-x64-dll.exe /VERYSILENT /TASKS=install_Devel
+  - SET PATH=C:\Ruby%RUBY_VERSION%\bin;C:\Program Files\ImageMagick-6.8.9-Q16;%PATH%
+  - bundle install
+
+build: off
```
Comment (mockdeep) [inline .appveyor.yml]: This means it's not actually running the tests, right?

File: ext/RMagick/rmimage.c [hunk 4]
```diff
@@ -4238,11 +4238,11 @@ Image_morphology_channel(VALUE self, VALUE channel_v, VALUE method_v, VALUE iter
   Data_Get_Struct(kernel_v, KernelInfo, kernel);
 
   image = rm_check_destroyed(self);
-  GetExceptionInfo(&exception);
+  exception = AcquireExceptionInfo();
```
Comment (mockdeep) [inline ext/RMagick/rmimage.c]: Would you mind adding a new pull request for changing `GetExceptionInfo` to `AcquireExceptionInfo`?

File: .appveyor.yml (hunk 3 above)
Comment (Watson1978) [reply .appveyor.yml]: We just run the tests using `test_script` section.
https://www.appveyor.com/docs/lang/ruby/

File: ext/RMagick/rmimage.c (hunk 4 above)
Comment (Watson1978) [reply ext/RMagick/rmimage.c]: OK, I see. I will split the commit.

File: .appveyor.yml (hunk 3 above)
Comment (mockdeep) [reply .appveyor.yml]: Oh, wow. So we've got a passing build. Woot!

File: ext/RMagick/rmimage.c [hunk 5]
```diff
@@ -9061,6 +9061,9 @@ Image_monitor_eq(VALUE self, VALUE monitor)
         (void) SetImageProgressMonitor(image, rm_progress_monitor, (void *)monitor);
     }
 
+#if defined(_WIN32)
+    rb_warn("Image#monitor= does not work on Windows");
+#endif
```
Comment (mockdeep) [inline ext/RMagick/rmimage.c]: Can we also pull this commit onto a separate PR? I've set up the commit settings to only allow squashing PRs, so it would be good to create separate PRs for different changes.

File: ext/RMagick/rmutil.c [hunk 6]
```diff
@@ -1445,6 +1445,7 @@ rm_progress_monitor(
     const MagickSizeType sp,
     void *client_data)
 {
+#if !defined(_WIN32)
```
Comment (mockdeep) [inline ext/RMagick/rmutil.c]: Can you explain what this is for?

Comment (Watson1978) [reply ext/RMagick/rmutil.c]: The SEGV has been occurred with old ImageMagick (v6.8.9) on only Windows.

If you set the `Proc` object through the `#monitor=` method, the object will be called each time ImageMagick processing progresses at https://github.com/rmagick/rmagick/blob/b1828e24af5eeb1e2b3c2f3e592552950a27bb6b/ext/RMagick/rmutil.c#L1464.

The Proc object is called correctly only once and SEGV always occurs in the second call.

I guess that there is a problem in callback processing with old ImageMagick and it breaks Proc object or something...

File: ext/RMagick/rmimage.c (hunk 5 above)
Comment (Watson1978) [reply ext/RMagick/rmimage.c]: 🆗

File: ext/RMagick/rmimage.c (hunk 4 above)
Comment (Watson1978) [reply ext/RMagick/rmimage.c]: https://github.com/rmagick/rmagick/pull/343

File: ext/RMagick/rmimage.c (hunk 5 above)
Comment (Watson1978) [reply ext/RMagick/rmimage.c]: https://github.com/rmagick/rmagick/pull/344

**Comment by mockdeep** [comment]
@Watson1978 okay, I think if you rebase this we should hopefully get a passing build again on Appveyor. Once we get that nailed down, I think we can go ahead and do a 3.0 release.

**Comment by Watson1978** [comment]
@mockdeep Thanks. rebased.
